Formula/ncurses.rb

```diff
@@ -23,15 +23,22 @@ class Ncurses < Formula
   end
 
   def install
-    system "./configure", "--prefix=#{prefix}",
-                          "--enable-pc-files",
-                          "--with-pkg-config-libdir=#{lib}/pkgconfig",
-                          "--enable-sigwinch",
-                          "--enable-symlinks",
-                          "--enable-widec",
-                          "--with-shared",
-                          "--with-gpm=no",
-                          "--without-ada"
+    args = [
+      "--prefix=#{prefix}",
+      "--enable-pc-files",
+      "--with-pkg-config-libdir=#{lib}/pkgconfig",
+      "--enable-sigwinch",
+      "--enable-symlinks",
+      "--enable-widec",
+      "--with-shared",
+      "--with-gpm=no",
+      "--without-ada",
+    ]
+    on_linux do
+      args << "--with-terminfo-dirs=/etc/terminfo:/lib/terminfo:/usr/share/terminfo"
+      args << "--with-default-terminfo-dir=/etc/terminfo"
```

Comment:
I think this means `ncurses` won't use its own `terminfo` database unless you set the environment variable `TERMINFO_DIRS=$(brew --prefix)/share/terminfo`. FYI.

`/usr/share/terminfo` probably works for Debian since that's where they install their `ncurses`.

Reply:
`/etc/terminfo` does not exist in the Linux docker CI. Not sure how bad it is to set this as the default then.

Reply:
Whoops; clicked on the wrong line. I was referring to `--with-terminfo-dirs`. I think you want to add `#{HOMEBREW_PREFIX}/share/terminfo` at the start of that.

Reply:
`/home/linuxbrew/.linuxbrew/share/terminfo` (e.g. #{HOMEBREW_PREFIX}/share/terminfo) does not exist, so adding it to the list of files does not change anything. One can use `infocmp -D` to check the paths used by ncurses to find terminfo stuff.

If we do nothing, this is what we have right now:
```
infocmp -D
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/current/share/terminfo
/home/linuxbrew/.linuxbrew/Cellar/ncurses/6.2/share/terminfo
```

With:
```
args << "--with-terminfo-dirs=/etc/terminfo:/lib/terminfo:/usr/share/terminfo"
args << "--with-default-terminfo-dir=/etc/terminfo"
```

we get:

```
infocmp -D
/etc/terminfo
/lib/terminfo
/usr/share/terminfo
```

If I remove `--with-default-terminfo-dir`, we get:

```
infocmp -D
/etc/terminfo
/lib/terminfo
/usr/share/terminfo
/home/linuxbrew/.linuxbrew/Cellar/ncurses/6.2/share/terminfo
```

I would like to remove the `--with-default-terminfo-dir`, as it looks like it's not doing anything special except removing the internal `/home/linuxbrew/.linuxbrew/Cellar/ncurses/6.2/share/terminfo` folder.

Reply:
Does `ncurses` not get linked into `HOMEBREW_PREFIX` on Linux? Even then, I think you want to add `#{share}/terminfo` at the start of `--with-terminfo-dirs`, as the terminfo database that ships with `ncurses` is likely to be newer than anything in `/{etc,lib,usr/share}/terminfo`.

Reply:
With the last version I pushed, the paths will be:

```
/home/linuxbrew/.linuxbrew/Cellar/ncurses/6.2/share/terminfo
/etc/terminfo
/lib/terminfo
/usr/share/terminfo
```

I think this is OK to be shipped.